docs/DeveloperGuide.md

```diff
@@ -2,8 +2,26 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+![Architecture Diagram](diagrams/ArchitectureDiagram.png) 
+
+The ***Architecture Diagram*** given above explains the high-level design of the App. 
+Given below is a quick overview of each component.
+
+**[`FridgeFriend`](https://github.com/AY2021S2-CS2113-T10-1/tp/blob/master/src/main/java/seedu/fridgefriend/FridgeFriend.java)** is driven by a method called `Main`. 
+It is responsible for,
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+[**`Food`**](#common-classes) represents a collection of classes used by multiple other components.
+
+The rest of the App consists of four components.
```

Comment:
Can we define our app as such:
- Food
- Command
- Exception
- Utilities
- FridgeFriend --> entry point

Reply:
Resolved